enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+  The kube-controller-manager logs may contain more details.)
+  reason: IngressControllerUnavailable
+  status: "False"
+  type: Available
+- lastTransitionTime: "2024-04-20T06:37:16Z"
+
+```
+
+##### Test when Ingress Controller CR has been provided with number of EIPs not equal to the number of public subnets for VPC tagged with your cluster name.
+  We will check the status of the IngressController which will have the message `The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: error creating load balancer: Must have same number of EIP AllocationIDs (4) and SubnetIDs (5)`
+  for status type `LoadBalancerReady` and `Available` in the Ingress Controller object as follows:
+```yaml
+- lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: |-
+   The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: error creating load balancer: Must have same number of EIP AllocationIDs (4) and SubnetIDs (5)
+   The kube-controller-manager logs may contain more details.
+  reason: SyncLoadBalancerFailed
+  status: "False"
+  type: LoadBalancerReady
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: LoadBalancer is not progressing
+  reason: LoadBalancerNotProgressing
+  status: "False"
+  type: LoadBalancerProgressing
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: The DNS management policy is set to Unmanaged.
+  reason: UnmanagedLoadBalancerDNS
+  status: "False"
+  type: DNSManaged
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: The wildcard record resource was not found.
+  reason: RecordNotFound
+  status: "False"
+  type: DNSReady
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: |-
+   One or more status conditions indicate unavailable: DeploymentAvailable=False (DeploymentUnavailable: The deployment has Available status condition set to False (reason: MinimumReplicasUnavailable) with message: Deployment does not have minimum availability.), LoadBalancerReady=False (SyncLoadBalancerFailed: The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: error creating load balancer: Must have same number of EIP AllocationIDs (4) and SubnetIDs (5)
+   The kube-controller-manager logs may contain more details.)
+  reason: IngressControllerUnavailable
+  status: "False"
+  type: Available
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: |-
+   One or more status conditions indicate progressing: DeploymentRollingOut=True (DeploymentRollingOut: Waiting for router deployment rollout to finish: 0 of 2 updated replica(s) are available...
+   )
+  reason: IngressControllerProgressing
+  status: "True"
+  type: Progressing
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+```
```

Comment:
Can you add a test to see what happens if a change was made to EIPs without the auto-delete annotation, and the controller goes into Pending, but then you add the auto-delete annotation?  Does it automatically delete at that point?

Reply:
no it does not. annotation should be present for automatic deletion.

Reply:
Hmm. I tested this in a cluster with my subnet annotation, and saw differently. I did see the service get automatically deleted.

Here's my findings: https://github.com/openshift/enhancements/pull/1595#discussion_r1596193017

Reply:
Hi,

Yes after adding the annotation, the service does get reflected with the desired eipAllocations. I can add a e2e test for it.